File: sqlbuilder/sqlbuilder/Equal.java
package com.clickability.cms.dataaccess.sqlbuilder;

public class Equal extends AbstractCompare<Equal> {

	public Equal() {
		super();
		this.using("=");
	}

	public Equal(Column lhs, Column rhs) {
		this();
		lhs(lhs);
		rhs(rhs);
	}

	public Equal(Column lhs) {
		this();
		lhs(lhs);
	}

	public Equal(String lhs) {
		this();
		lhs(lhs);
	}

	public Equal(String lhs, String rhs) {
		this();
		lhs(lhs);
		rhs(rhs);
	}

	public Equal(String lhs, int rhs) {
		this();
		lhs(lhs);
		rhs(rhs);
	}
}

File: queryBuilder/src/main/kotlin/com/inunison/lib/queryBuilder/Equal.kt
package com.inunison.lib.queryBuilder

internal class Equal() : AbstractCompare<Equal>() {

    init {
        this.using("=")
    }

    constructor(lhs: Column, rhs: Column) : this() {
        lhs(lhs)
        rhs(rhs)
    }

    constructor(lhs: Column) : this() {
        lhs(lhs)
    }

    constructor(lhs: String) : this() {
        lhs(lhs)
    }

    constructor(lhs: String, rhs: String) : this() {
        lhs(lhs)
        rhs(rhs)
    }

    constructor(lhs: String, rhs: Int) : this() {
        lhs(lhs)
        rhs(rhs)
    }
}
